tests/interface/test_cli.py

```diff
@@ -0,0 +1,50 @@
+from pathlib import Path
+
+import pytest
+import yaml
+from click.testing import CliRunner
+
+from tests import HARRY_POTTER_CONFIG
+from vivarium.interface.cli import simulate
+
+
+@pytest.fixture
+def runner():
+    return CliRunner()
+
+
+@pytest.fixture
+def model_spec(base_config, tmp_path) -> str:
+    base_config.update(HARRY_POTTER_CONFIG)
+    model_spec = {}
+    model_spec["configuration"] = base_config.to_dict()
+    model_spec["components"] = {
+        "tests": [
+            "Hogwarts()",
+            "HousePointsObserver()",
+            "NoStratificationsQuidditchWinsObserver()",
+            "QuidditchWinsObserver()",
+            "HogwartsResultsStratifier()",
+        ],
+    }
+    filepath = f"{tmp_path}/model_spec.yaml"
+    with open(filepath, "w") as f:
+        yaml.dump(model_spec, f)
+    return filepath
+
+
+def test_simulate_run(runner, model_spec):
+    output_dir = "/".join(model_spec.split("/")[:-1])
+    sim = runner.invoke(simulate, ["run", model_spec, "-o", output_dir])
+    assert sim.exit_code == 0
+    results_dir = list(Path(output_dir).rglob("*/simulation.log"))[0].parent
+    assert {file.name for file in results_dir.rglob("*.parquet")} == {
+        "house_points.parquet",
+        "quidditch_wins.parquet",
+        "no_stratifications_quidditch_wins.parquet",
+    }
+    with open(results_dir / "metadata.yaml", "r") as f:
+        metadata = yaml.safe_load(f)
+
+    assert set(metadata) == set(["input_draw", "random_seed", "simulation_run_time"])
+    assert metadata["simulation_run_time"] > 0.0
```

Comment:
Since you're here, why not also test the other flags of `simulate run`, like `-i`.

Reply:
@rmudambi Can you expand what you mean? Are you proposing doing the same test but w/ an actual test artifact (`-i`) and/or a location defined (`-l`) and the corresponding artifact? These tests don't actually use an artifact so those options weren't readily testable.

Reply:
You could mock one though?

Reply:
@rmudambi How is this? It kinda feels like cheating to just add to metadata, but I also feel like this kind of stuff is usesful for each sim to save out anyway?

Reply:
Does the model spec get dumped somewhere? Checking the artifact key in the model spec would be a better test if you can do that.

Reply:
I'm going to go ahead and merge this and then update to use the artifact key in the model spec in another PR that saves out the model spec